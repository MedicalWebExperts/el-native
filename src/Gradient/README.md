Gradient Component

This component provides a wrapper to other components so that a vertical linear
gradient is shown as background.

It expects:

children: Elements to be nested inside the gradient.
gradientColors (required): An array of colors to form the gradient.
style: Provided to override the size and shape of the container element.

If not a vertical gradient is wanted, index.js must be modified, being start, end
and locations the attributes responsible for this behavior.

IMPORTANT:
This component is based on 'react-native-linear-gradient'. That means you have to install
it as a dependency in your project in order to work.

For more information reach https://github.com/react-native-community/react-native-linear-gradient
