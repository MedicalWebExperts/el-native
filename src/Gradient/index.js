import React from 'react';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import LinearGradient from 'react-native-linear-gradient';

import styles from './styles';

const propTypes = {
  children: PropTypes.node,
  gradientColors: PropTypes.array.isRequired,
  locations: PropTypes.array.isRequired,
  style: PropTypes.object,
};

const defaultProps = {
  children: undefined,
  gradientColors: [],
  locations: [0.3, 0.6, 1],
  style: {},
};

// locations has to have the same amount as colors, if not a warning is raised

function Gradient({ children, gradientColors, locations, style }) {
  const stylesCommon = style ? StyleSheet.flatten([styles, style]) : styles;
  const gradient = (
    <LinearGradient
      colors={gradientColors}
      start={{ x: 0, y: 1 }}
      end={{ x: 0.8, y: 1 }}
      locations={locations}
      style={stylesCommon.container}
    >
      {children}
    </LinearGradient>
  );
  return gradient;
}

Gradient.propTypes = propTypes;
Gradient.defaultProps = defaultProps;

export default Gradient;
