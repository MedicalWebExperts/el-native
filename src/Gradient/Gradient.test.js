import React from 'react';
import renderer from 'react-test-renderer';

import Gradient from './index';

const defaultProps = {
  gradientColors: ['#FFFFFF', '#FAFAFA', '#EAEAEA'],
};

describe('Snapshots', () => {
  it('should render default gradient', () => {
    const component = renderer.create(<Gradient
      {...defaultProps}
    />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
