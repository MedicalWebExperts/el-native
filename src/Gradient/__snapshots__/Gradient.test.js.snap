// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should render default gradient 1`] = `
<BVLinearGradient
  colors={
    Array [
      4294967295,
      4294638330,
      4293585642,
    ]
  }
  end={
    Object {
      "x": 0.8,
      "y": 1,
    }
  }
  locations={
    Array [
      0.3,
      0.6,
      1,
    ]
  }
  start={
    Object {
      "x": 0,
      "y": 1,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "height": 64,
      "justifyContent": "center",
    }
  }
/>
`;
